public/app/plugins/panel/geomap/layers/data/dayNightLayer.tsx

```diff
@@ -0,0 +1,260 @@
+          const time = event.payload?.point?.time as number;
+          if (time) {
+            const lineTime = new Date(time);
+            const nightLinePoints = sourceLine.getCoordinates(lineTime.toString(), 'line');
+            nightLineLayer.getSource()?.clear();
+            const lineStringArray:Coordinate[][] = [];
+            for (let l = 0; l < nightLinePoints.length - 1; l++){
+              const x1:number = Object.values(nightLinePoints[l])[0];
+              const y1:number = Object.values(nightLinePoints[l])[1];
+              const x2:number = Object.values(nightLinePoints[l+1])[0];
+              const y2:number = Object.values(nightLinePoints[l+1])[1];
+              const lineString = [fromLonLat([x1, y1]),fromLonLat([x2, y2])];
+              lineStringArray.push(lineString);
+            }
+            nightLineLayer.getSource()?.addFeature(new Feature({
+                geometry: new MultiLineString(lineStringArray),
+              }))
+
+            let sunLinePos: number[] = [];
+            sunLinePos = sourceLineMethods.getSunPosition(lineTime);
+            sunLineFeature.getGeometry()?.setCoordinates(fromLonLat(sunLinePos));
+            sunLineFeature.setStyle([sunLineStyle, sunLineStyleDash]);
+          }
+        })
+      );
+
+      subscriptions.add(
+        eventBus.subscribe(DataHoverClearEvent, (event) => {
+          nightLineLayer.getSource()?.clear();
+          sunLineFeature.setStyle(new Style({}));
+        })
+      );
+    }
+
+    return {
+      init: () => layer,
+      dispose: () => subscriptions.unsubscribe(),
+      update: (data: PanelData) => {
+        const from = new Date(data.timeRange.from.valueOf());
+        const to = new Date(data.timeRange.to.valueOf());
+        let selectedTime: Date = new Date();
+        let sunPos: number[] = [];
+        // TODO: add option for "Both"
+        if (config.show === ShowTime.From){
+          selectedTime = from;
+        } else {
+          selectedTime = to;
+        }
+
+        source.setTime(selectedTime);
+        if (config.sun){
+          sunPos = sourceMethods.getSunPosition(selectedTime);
+          sunFeature.getGeometry()?.setCoordinates(fromLonLat(sunPos));
+        }
+      },
+
+      // Marker overlay options
+      registerOptionsUI: (builder) => {
+        builder
+          .addRadio({
+            path: 'config.show',
+            name: 'Show',
+            settings: {
+              options: [
+                { label: 'From', value: ShowTime.From },
+                { label: 'To', value: ShowTime.To },
+              ],
+            },
+            defaultValue: defaultConfig.show,
+          });
+        builder.addColorPicker({
+          path: 'config.nightColor',
+          name: 'Night region color',
+          description: 'Pick color of night region',
+          defaultValue: 'rgb(0,0,0)',
+          settings: [{enableNamedColors: false}],
+        });
```

Comment:
This currently doesn't have the best UX, especially in dark mode
- hard to see the color picker UI when default color
- User has to click on small color picker circle while clicking on rest of field does nothing
<img width="708" alt="Screen Shot 2022-06-17 at 8 12 52 PM" src="https://user-images.githubusercontent.com/22381771/174421049-ceaa42b1-3866-4b3a-9a9c-b04ba11d156e.png">

Reply:
Looks like addressing underlying editor UX is a slightly larger task so for now an easy stop gap is to just change default night color to stand out better for dark mode

<img width="964" alt="Screen Shot 2022-06-18 at 1 52 18 PM" src="https://user-images.githubusercontent.com/22381771/174456702-02d5dc01-3c4b-4dd9-acfb-27dcef942536.png">